docs/index.rst

```diff
@@ -111,54 +113,75 @@ function.
 
 .. code-block:: python
 
+    import flask
+    from flask_login import login_user
+
     @app.route('/login', methods=['GET', 'POST'])
     def login():
         # Here we use a class of some kind to represent and validate our
         # client-side form data. For example, WTForms is a library that will
         # handle this for us, and we use a custom LoginForm to validate.
         form = LoginForm()
+
         if form.validate_on_submit():
             # Login and validate the user.
             # user should be an instance of your `User` class
             login_user(user)
 
             flask.flash('Logged in successfully.')
 
-            next = flask.request.args.get('next')
+            next_url = form.next_url.data
             # is_safe_url should check if the url is safe for redirects.
             # See http://flask.pocoo.org/snippets/62/ for an example.
-            if not is_safe_url(next):
+            if not is_safe_url(next_url):
                 return flask.abort(400)
 
-            return flask.redirect(next or flask.url_for('index'))
+            return flask.redirect(next_url or flask.url_for('index'))
+
+        # Pass the next URL along through the Login form.
+        form.next_url.data = flask.request.args.get('next')
```

Comment:
This is not good practice. You should never set form data after validation, and the next url isn't part of the LoginForm conceptually so the previous way was more correct imo.

Reply:
If [this comment](https://github.com/maxcountryman/flask-login/pull/416/files/8a4a95cc329aff7070ebcc9f508c06dbb4778513#r228954737) is resolved then this change will no longer be necessary.